test/integration/puppetlabs/services/legacy_routes/legacy_routes_test.clj

```diff
@@ -31,20 +31,22 @@
 
 (deftest ^:integration legacy-routes
   (testing "The legacy web routing service properly handles old routes."
-    (bootstrap/with-puppetserver-running app
-      {}
-      (is (= 200 (:status (http-get "/v2.0/environments"))))
-      (is (= 200 (:status (http-get "/production/node/localhost"))))
-      (is (= 200 (:status (http-get "/production/certificate_statuses/all")))))))
+    (bootstrap/with-puppetserver-running-with-mock-jrubies
+     app
+     {}
+     (is (= 200 (:status (http-get "/v2.0/environments"))))
+     (is (= 200 (:status (http-get "/production/node/localhost"))))
+     (is (= 200 (:status (http-get "/production/certificate_statuses/all")))))))
```

Comment:
It might.

I haven't finished reading the rest of your comments yet.  I'd still like to know how many tests would go red if we have that requesthandler thing throw an exception, and which ones they are.  Maybe you've talked about that elsewhere.